references/video_classification/train.py

```diff
@@ -0,0 +1,340 @@
+
+    print("Loading training data")
+    st = time.time()
+    cache_path = _get_cache_path(traindir)
+    dataset = torch.load("/private/home/fmassa/github/vision/kinetics_train.pth")
+    # dataset = torch.load("/private/home/fmassa/github/vision/kinetics_val.pth")
+    dataset.video_clips.compute_clips(16, 1, frame_rate=15)
+    dataset.transform = torchvision.transforms.Compose([
+        T.ToFloatTensorInZeroOne(),
+        T.Resize((128, 171)),
+        T.RandomHorizontalFlip(),
+        T.Normalize((0.43216, 0.394666, 0.37645), (0.22803, 0.22145, 0.216989)),
+        T.RandomCrop((112, 112))
+    ])
+
+    """
+    if args.cache_dataset and os.path.exists(cache_path):
+        # Attention, as the transforms are also cached!
+        print("Loading dataset_train from {}".format(cache_path))
+        dataset, _ = torch.load(cache_path)
+    else:
+        dataset = torchvision.datasets.Kinetics(
+            traindir,
+            transforms.Compose([
+                transforms.RandomResizedCrop(224),
+                transforms.RandomHorizontalFlip(),
+                transforms.ToTensor(),
+                normalize,
+            ]))
+        if args.cache_dataset:
+            print("Saving dataset_train to {}".format(cache_path))
+            utils.mkdir(os.path.dirname(cache_path))
+            utils.save_on_master((dataset, traindir), cache_path)
+    """
+    print("Took", time.time() - st)
+
+    print("Loading validation data")
+    cache_path = _get_cache_path(valdir)
+    dataset_test = torch.load("/private/home/fmassa/github/vision/kinetics_val.pth")
+    dataset_test.video_clips.compute_clips(16, 1, frame_rate=15)
+    dataset_test.transform = torchvision.transforms.Compose([
+        T.ToFloatTensorInZeroOne(),
+        T.Resize((128, 171)),
+        T.Normalize((0.43216, 0.394666, 0.37645), (0.22803, 0.22145, 0.216989)),
+        T.CenterCrop((112, 112))
+    ])
+    """
+    if args.cache_dataset and os.path.exists(cache_path):
+        # Attention, as the transforms are also cached!
+        print("Loading dataset_test from {}".format(cache_path))
+        dataset_test, _ = torch.load(cache_path)
+    else:
+        dataset_test = torchvision.datasets.Kinetics(
+            valdir,
+            args.frames_per_clip,
+            1,
+            transforms.Compose([
+                transforms.Resize(256),
+                transforms.CenterCrop(224),
+                transforms.ToTensor(),
+                normalize,
+            ]))
+        if args.cache_dataset:
+            print("Saving dataset_test to {}".format(cache_path))
+            utils.mkdir(os.path.dirname(cache_path))
+            utils.save_on_master((dataset_test, valdir), cache_path)
+    """
+    print("Creating data loaders")
+    train_sampler = torchvision.datasets.video_utils.RandomClipSampler(dataset.video_clips, args.clips_per_video)
+    test_sampler = SequentialClipSampler(dataset_test.video_clips, args.clips_per_video)
```

Comment:
Do we plan to add this and the uniform sampler to the `video_utils`?

Reply:
Yes, I didn't add it there because I'm not sure I like the place where the current sampler is. I'm actually thinking about moving the other sampler away from there, and inside `references` for now, while we decide where to put it

Reply:
I'll move `RandomClipSampler` in a follow-up PR